src/SecretManager/Microsoft.DncEng.SecretManager/SecretTypes/DomainAccount.cs

```diff
@@ -30,24 +31,25 @@ protected override async Task<SecretData> RotateValue(Parameters parameters, Rot
             throw new HumanInterventionRequiredException($"User intervention required for creation or rotation of a Domain Account.");
         }
 
-        string generatedPassword = PasswordGenerator.GenerateRandomPassword(20, false);
         string password = await context.GetSecretValue(new SecretReference(context.SecretName));
         if (!string.IsNullOrEmpty(password))
             _console.WriteLine($"Current password for account {parameters.AccountName}: {password}");
 
         _console.WriteLine($@"Steps:
-1. Ctrl-alt-delete on a domain joined windows computer
-2. Put in the name of the domain account {parameters.AccountName}
-3. Put the current secret {password} into the ""Old Password""
-4. Put the new password {generatedPassword} or your custom one in the ""New Password"" field
-5. Update the account");
+1. Visit https://coreidentity.microsoft.com/manage/service and review expiration dates for this bot account.
+   Will need to click on the name link (e.g. https://coreidentity.microsoft.com/manage/Service/redmond/dn-bot)
+   for the bot account to see both password and account expiration dates.
+2. Click on 'Extend' if the account itself is expired or will expire soon.
+3. Click on 'Reset Password' if the password is expired or will expire soon.
+4. Copy the generated password to the clipboard before doing anything else. Paste it when requested in step 5.
+5. Run 'secret-manager --force-secret={parameters.AccountName}' to update the key vault with the new password.");
 
         if (!string.IsNullOrWhiteSpace(parameters.Description))
             _console.WriteLine($"Additional information: {parameters.Description}");
 
-        string newPassword = await _console.PromptAsync($"Enter a new password or press enter to use a generated password {generatedPassword} : ");
-        if (string.IsNullOrWhiteSpace(newPassword))
-            newPassword = generatedPassword;
+        string newPassword = await _console.PromptAndValidateAsync("New Password",
+            "Expecting a valid password from CoreIdentity, containing at least 20 fairly random characters",
+            value => value != null && value.Length >= 20 && !value.All(char.IsLetterOrDigit));
 
         return new SecretData(newPassword, DateTimeOffset.MaxValue, _clock.UtcNow.AddMonths(6));
```

Comment:
This should be updated, too. The password expiration is enforced, and the rotation time should be a bit before that expiration (perhaps even just before the email gets sent, then the email serves as a "something's wrong" check and also doesn't cause confusion when both services are messaging a rotation).

Reply:
I didn't do the math to figure out when CoreIdentity decides to warn us. main thing I know is the last round of emails arrived b/4 `secret-manager` started complaining and the new expirations are about 6 months out. I'll figure the right `Add...` values b/4 merging this

Reply:
```suggestion
        return new SecretData(newPassword, DateTimeOffset.MaxValue, _clock.UtcNow.AddMonths(5));
```
Just looked at the history, I think 5 months will be a good rotation time.